auto_nag/scripts/needinfo_regression_author.py

```diff
@@ -97,6 +97,16 @@ def bug_handler(regressor_bug):
         ).get_data().wait()
 
     def filter_bugs(self, bugs):
+        # Exclude bugs whose regressor author is nobody.
+        for bug in list(bugs.values()):
+            if utils.is_no_assignee(bug["regressor_author_email"]):
+                logger.warning(
+                    "Bug {}, regressor of bug {}, doesn't have an author".format(
+                        bug["regressor_id"], bug["id"]
+                    )
+                )
+                del bugs[bug["id"]]
+
         # Exclude bugs whose creator is the regressor author.
         bugs = {
```

Comment:
This bug filtering process could be merged with the one you just introduced.  Finale version could look like the following:
```py
        for bug in bugs.values():
            # Exclude bugs whose regressor author is nobody.
            if utils.is_no_assignee(bug["regressor_author_email"]):
                logger.warning(
                    "Bug {}, regressor of bug {}, doesn't have an author".format(
                        bug["regressor_id"], bug["id"]
                    )
                )
                del bugs[bug["id"]]

            # Exclude bugs whose creator is the regressor author.
            elif bug["creator"] == bug["regressor_author_email"]:
                del bugs[bug["id"]]
```

Reply:
I prefer using the comprehensions when possible, it feels more Pythonic. I didn't use it here just so I could log the warning.